cognito-csv-exporter.js

```diff
@@ -0,0 +1,79 @@
+const createCsvWriter = require('csv-writer').createArrayCsvWriter;
+const { userToCsvHeader } = require('./csv-utils/userToCsv');
+const AWS = require('aws-sdk');
+
+require('dotenv').config();
+
+// Configure the AWS SDK with your credentials
+AWS.config.update({
+  accessKeyId: 'your_access_key',
+  secretAccessKey: 'your_secret_access_key',
+  region: 'your_region',
+});
+
+// Create an instance of the AWS Cognito Identity provider client
+const cognitoIdentityServiceProvider = new AWS.CognitoIdentityServiceProvider();
+const userPoolId = 'user-pool-id';
+
+const getAllUsers = async (params) => {
+  try {
+    // string must not be empty
+    let paginationToken = 'notEmpty';
+    let itemsAll = {
+      Users: [],
+    };
+    while (paginationToken) {
+      const data = await cognitoIdentityServiceProvider
+        .listUsers(params)
+        .promise();
+
+      const { Users } = data;
+      itemsAll = {
+        ...data,
+        ...{ Users: [...itemsAll.Users, ...(Users ? [...Users] : [])] },
+      };
+      paginationToken = data.PaginationToken;
+      if (paginationToken) {
+        params.PaginationToken = paginationToken;
+      }
+    }
+    return itemsAll;
+  } catch (err) {
+    console.error('Unable to scan the cognito pool users. Error JSON:', err);
+  }
+};
+
+const init = async () => {
+  const params = {
+    UserPoolId: userPoolId,
+  };
+
+  const allUsers = await getAllUsers(params);
+
+  // provide the user pool ID of the new pool you want to transfer users to
+  cognitoIdentityServiceProvider.getCSVHeader(
+    {
+      UserPoolId: 'example-user-pool-id',
```

Comment:
For convenience, I would suggest moving all of these strings that are meant to be replaced by the user into consts up at the top of the file. I would use ALL_CAPS for them, and make it obvious that their values are meant to be set manually before running the script

Reply:
Good idea. I implemented this.